### ISSUE TITLE
Catch-All Email Broken

### ISSUE THREAD

**Issue by dribgnikcom** [virtualmin/virtualmin-gpl#578]
For many months I pondered the issue of the catch-all account catching everything, even mails sent to unique addresses, and then I stumbled upon this interesting tidbit posted here:

**issue #216** 

_Rant, but not at Virtualmin... Using "user@domain.com" format for everything and the "Create extra Unix user on Postfix systems when using user@domain format?"_

I had the idea that this fix, implemented relatively recently, was what was causing this regression, I wanted to to test whether undoing it would solve the issue.

So I commented out "resolve_dequoted_address = no" from /etc/postfix/main.cf and set "Create extra Unix user on Postfix systems when using user@domain format" back to yes (from the now default "no") in Virtualmin and proceeded to test it and it works fine, with the designated account catching all the stray emails, while other accounts still receiving email unique to their address (whereas beforehand, **_everything_** would go to the catch-all).

So, in conclusion, it is necessary to revert to the old behavior if you need this feature.  If you can fix this and still incorporate the fix from #216, great.  If not, then no big deal for me.  I understand why you stress the importance of not doing things the old way with the user creation, but this is primarily an email server, nothing else.

**Comment by iliajie** [comment]
Hello,

Are you seeing this with the latest Virtualmin 7.7? Is this a clean new install or an old system?

**Comment by dribgnikcom** [comment]
Yes sir, brand new install, new system, Ubuntu 22.04.  Tested and re-tested.  I reckon it would be very easy for you to reproduce.

**Comment by dribgnikcom** [comment]
This might be a longshot, but I would recommend switching the default from Postfix to Sendmail.  Postfix just does not work properly.  Within minutes I had sendmail doing relaying properly without any configuration file hairsplitting or otherwise.  Unfortunately, the Sendmail implementation in Virtualmin is primitive and I could not get SMTP to function.

I know this isn't going to happen, but like I said, it's a longshot.  Maybe at least a switch for virtualmin.sh to install with sendmail instead of that confounded postfix.

**Comment by swelljoe** [comment]
We will definitely not switch to Sendmail as the default MTA in Virtualmin, but Sendmail is well-supported by Virtualmin.

You'd be mostly on your own in terms of getting support in our forums, etc., but if you find bugs, you're welcome to file them and they're likely to get fixed.

**Comment by dribgnikcom** [comment]
I can no longer replicate the problem.  It's 'just working' for me now.  I address what I did differently in this thread:

https://forum.virtualmin.com/t/how-to-disable-aliases-feature-completely/121345/17

But I don't see why that fixed it (maybe switching to Debian helped?).

**Comment by amitsamanta996** [comment]
catch all is still broken

**Comment by iliajie** [comment]
> catch all is still broken

@amitsamanta996 A few questions:

  1. Do you have Virtualmin 7.7 installed?
  2. Did you install Virtualmin using `virtualmin-install.sh` script?
  3. Do you have creation of extra Postfix user enabled in Virtualmin cnofiguration?

**Comment by iliajie** [comment]
> If you want catch-all for the domain which is same as hostname, then the extra stuff needs to happen

@swelljoe Highly recommended against enabling mail for the hostname. Joe, could you explain one more time why enabling mail for the hostname is discouraged?
